### DISCUSSION TITLE
How to disable lazygit in LazyVim?

### DISCUSSION THREAD

**Discussion by moonstruckmoth** [LazyVim/LazyVim#3152] [Q&A] [answered]
I use fugitive for my git tasks and I would like to disable lazygit.
I could not find any way to disable it cleanly.

TIA

**Comment by dpetka2001** [comment]
Lazygit is just a cli tool. If you don't have it installed Lazyvim won't use it. If you want to use the same keymaps for fugitive, just change the keymaps to thes you want in `/lua/config/keymaps.lua`.

**Reply by moonstruckmoth** [reply]
I understand lazygit is a cli tool and I have used it. I liked it but I prefer fugitive

The reason I asked if lazygit configuration can be disabled is I had added keymaps for fugitive along with the plugin configuration but these (atlleast those which lazygit uses) keymaps get overwritten by lazygit keymaps unless they're overridden in keymaps.lua

I would have preferred if my keymap settings for fugitive were kept with plugin config.

**Reply by dpetka2001** [reply]
Can you show how you're setting up fugitive keymaps?

**Reply by moonstruckmoth** [reply]
Thanks for ur reply. 
Also, this is not super urgent, just something which was bothering me...

From my lua/plugins/editor.lua

```
{
    "tpope/vim-fugitive",
    event = "VeryLazy",
    config = function()
      map("n", "<leader>gb", "<cmd>Git blame<cr>", { desc = "Git blame" })
      map("n", "<leader>gC", "<cmd>Gdiffsplit!<cr>", { desc = "Conflict 3-way split" })
      map("n", "<leader>gd", "<cmd>Gdiff<cr>", { desc = "Gdiff" })
      map("n", "<leader>gD", "<cmd>Git log --stat -p<cr>", { desc = "Git log --stat -p" })
      map("n", "<leader>ge", ":Gedit ", { desc = "Gedit" }) -- Gedit can take commit objects
      map("n", "<leader>gl", [[<cmd>Git log --format="%h [%ad] [%an] %s"<cr>]], { desc = "Git log oneline" })
      map("n", "<leader>gL", "<cmd>Git log<cr>", { desc = "Git log" })
      map("n", "<leader>gg", "<cmd>G<cr>", { desc = "G" })
      map("n", "<leader>gP", "<cmd>Git pull", { desc = "Git pull" })
      map("n", "<leader>gp", "<cmd>Git -c push.default=current push<cr>", { desc = "Git -c push.default=current push" })
      map("n", "<leader>gr", "<cmd>Gread<cr>", { desc = "Gread" })
      map("n", "<leader>gw", "<cmd>Gwrite<cr>", { desc = "Gwrite" })
      map("n", "<leader>gu", "<cmd>diffupdate<cr>", { desc = "diffupdate" })
      map("n", "<leader>g2", "<cmd>diffget //2<cr>", { desc = "diffget //2" })
      map("n", "<leader>g3", "<cmd>diffget //3<cr>", { desc = "diffget //3" })
    end,
  },
```

If I add any keymap used by lazygit in the above config, it gets overridden.
One way around it is to add it in keymaps.lua

From my lua/config/keymaps.lua

`map("n", "<leader>gg", "<cmd>G<cr>", { desc = "G" })`

The lazygit keymaps which are not overridden or deleted in keymaps.lua still show up.
For eg, I've not overridden `<leader>gG`. Such keymaps show up

**Reply by dpetka2001** [reply]
Try the following
```lua
return {
  {
    "tpope/vim-fugitive",
    event = "VeryLazy",
    config = function()
      vim.schedule(function()
        vim.keymap.set("n", "<leader>gb", "<cmd>Git blame<cr>", { desc = "Git blame" })
        vim.keymap.set("n", "<leader>gC", "<cmd>Gdiffsplit!<cr>", { desc = "Conflict 3-way split" })
        vim.keymap.set("n", "<leader>gd", "<cmd>Gdiff<cr>", { desc = "Gdiff" })
        vim.keymap.set("n", "<leader>gD", "<cmd>Git log --stat -p<cr>", { desc = "Git log --stat -p" })
        vim.keymap.set("n", "<leader>ge", ":Gedit ", { desc = "Gedit" }) -- Gedit can take commit objects
        vim.keymap.set("n", "<leader>gl", [[<cmd>Git log --format="%h [%ad] [%an] %s"<cr>]], { desc = "Git log oneline" })
        vim.keymap.set("n", "<leader>gL", "<cmd>Git log<cr>", { desc = "Git log" })
        vim.keymap.set("n", "<leader>gg", "<cmd>G<cr>", { desc = "G" })
        vim.keymap.set("n", "<leader>gP", "<cmd>Git pull", { desc = "Git pull" })
        vim.keymap.set("n", "<leader>gp", "<cmd>Git -c push.default=current push<cr>", { desc = "Git -c push.default=current push" })
        vim.keymap.set("n", "<leader>gr", "<cmd>Gread<cr>", { desc = "Gread" })
        vim.keymap.set("n", "<leader>gw", "<cmd>Gwrite<cr>", { desc = "Gwrite" })
        vim.keymap.set("n", "<leader>gu", "<cmd>diffupdate<cr>", { desc = "diffupdate" })
        vim.keymap.set("n", "<leader>g2", "<cmd>diffget //2<cr>", { desc = "diffget //2" })
        vim.keymap.set("n", "<leader>g3", "<cmd>diffget //3<cr>", { desc = "diffget //3" })
      end)
    end,
  },
}
```

**Reply by moonstruckmoth** [reply]
Thanks for ur quick reply.

This is working, I'm able to override even lazygit keymaps now. 

I tried adding `vim.keymap.del("n", "<leader>gG")` inside, but this doesn't work.

It didn't work even inside keymaps.lua.

**Reply by dpetka2001** [reply, marked as answer]
When I delete the keymap in `~/.config/nvim/lua/config/keymaps.lua` it doesn't show up any more. Keep your fugitive keymaps inside the fugitive config function and disable the ones you don't want globally in the `keymaps.lua` file.

**Reply by moonstruckmoth** [reply]
It worked when I moved the delete calls to the top of my keymaps file.
I will investigate this weird behavior.

But thanks again for taking the time to look into this.
